packages/scripts/src/lib/_migrate-glintrc.ts

```diff
@@ -0,0 +1,302 @@
+import { readFile, writeFile } from 'fs/promises';
+import * as path from 'path';
+
+import { evaluate, patch } from 'golden-fleece';
+import yaml from 'js-yaml';
+import Result, { err, isInstance, ok, tryOrElse } from 'true-myth/result';
+import Unit from 'true-myth/unit';
+import yargs from 'yargs';
+
+import { z } from 'zod';
+
+const EnvironmentList = z.array(z.string());
+
+const EnvironmentMap = z.record(
+  z.object({
+    additionalGlobals: z.optional(z.array(z.string())),
+  })
+);
+type EnvironmentMap = z.infer<typeof EnvironmentMap>;
+
+const Environment = z.optional(z.union([z.string(), EnvironmentList, EnvironmentMap]));
+const Files = z.optional(z.array(z.string()));
+
+const GlintRc = z.object({
+  environment: Environment,
+  checkStandaloneTemplates: z.optional(z.boolean()),
+  include: Files,
+  exclude: Files,
+});
+
+type GlintRc = z.infer<typeof GlintRc>;
+
+type GlintConfig = {
+  environment?:
+    | string
+    | string[]
+    | Record<
+        string,
+        {
+          additionalGlobals?: string[] | undefined;
+        }
+      >;
+  checkStandaloneTemplates?: boolean;
+  transform?: {
+    include?: string[];
+    exclude?: string[];
+  };
+};
+
+function loadFile(path: string): Promise<Result<string, string>> {
+  return readFile(path, { encoding: 'utf-8' })
+    .then((v) => ok<string, string>(v))
+    .catch((e) => err(`Could not load file at ${path}: ${JSON.stringify(e)}`));
+}
+
+function loadOrCreateTsconfig(configPath: string): Promise<string> {
+  return readFile(configPath, { encoding: 'utf-8' }).catch((e) => {
+    console.info(
+      `Could not load tsconfig.json at ${configPath}: ${e}; attempting to create a new one`
+    );
+    return '{}';
+  });
+}
```

Comment:
I *think* this is the behavior we want: if there is a `.glintrc.yml` somewhere and there is *not* a corresponding `tsconfig.json`, we will create one, and the only thing it will have there is a `glint: { ... }` configuration. Does that seem right?

Reply:
That seeeeems fine? In practice in < 0.9 I think we effectively required your `.glintrc` and `tsconfig` to be colocated with one another or things would break in weird ways (which was part of the point of this migration—acknowledging and embracing that coupling)

Reply:
I'll go with this and if it proves a problem we can fix it up.